tests/model/test-msh.cpp

```diff
@@ -114,26 +114,110 @@ void test_brep( const geode::BRep& brep )
+        brep.nb_surfaces() == 12, "[Test] Number of surfaces is not correct" );
+    OPENGEODE_EXCEPTION(
+        brep.nb_blocks() == 4, "[Test] Number of blocks is not correct" );
+
+    // Number of vertices and elements in components
+    for( const auto& c : brep.corners() )
+    {
+        OPENGEODE_EXCEPTION( c.mesh().nb_vertices() == 1,
+            "[Test] Number of vertices in corners should be 1" );
+    }
+    for( const auto& l : brep.lines() )
+    {
+        OPENGEODE_EXCEPTION( l.mesh().nb_vertices() > 0,
+            "[Test] Number of vertices in lines should not be null" );
+        OPENGEODE_EXCEPTION( l.mesh().nb_edges() > 0,
+            "[Test] Number of edges in lines should not be null" );
+    }
+    for( const auto& s : brep.surfaces() )
+    {
+        OPENGEODE_EXCEPTION( s.mesh().nb_vertices() > 0,
+            "[Test] Number of vertices in surfaces should not be null" );
+        OPENGEODE_EXCEPTION( s.mesh().nb_polygons() > 0,
+            "[Test] Number of polygons in surfaces should not be null" );
+    }
+
+    for( const auto& b : brep.blocks() )
+    {
+        OPENGEODE_EXCEPTION( b.mesh().nb_vertices() > 0,
+            "[Test] Number of vertices in blocks should not be null" );
+        OPENGEODE_EXCEPTION( b.mesh().nb_polyhedra() > 0,
+            "[Test] Number of polyhedra in blocks should not be null" );
+    }
+
+    // Number of component boundaries and incidences
+    for( const auto& c : brep.corners() )
+    {
+        OPENGEODE_EXCEPTION( brep.nb_boundaries( c.id() ) == 0,
+            "[Test] Number of corner boundary should be 0" );
+        OPENGEODE_EXCEPTION( brep.nb_incidences( c.id() ) == 4
+                                 || brep.nb_incidences( c.id() ) == 5,
+            "[Test] Number of corner incidences should be 4 or 5" );
+    }
+    for( const auto& l : brep.lines() )
+    {
+        OPENGEODE_EXCEPTION( brep.nb_boundaries( l.id() ) == 2,
+            "[Test] Number of line boundary should be 2" );
+        OPENGEODE_EXCEPTION( brep.nb_incidences( l.id() ) > 1
+                                 && brep.nb_incidences( l.id() ) < 5,
+            "[Test] Number of line incidences should be 2, 3 or 4" );
+    }
+    for( const auto& s : brep.surfaces() )
+    {
+        OPENGEODE_EXCEPTION( brep.nb_boundaries( s.id() ) == 3,
+            "[Test] Number of surface boundary should be 3" );
+        OPENGEODE_EXCEPTION( brep.nb_incidences( s.id() ) == 1
+                                 || brep.nb_incidences( s.id() ) == 2,
+            "[Test] Number of surface incidences should be 1 or 2" );
+    }
+    for( const auto& b : brep.blocks() )
+    {
+        OPENGEODE_EXCEPTION( brep.nb_boundaries( b.id() ) == 4,
+            "[Test] Number of block boundary should be 4" );
+        OPENGEODE_EXCEPTION( brep.nb_incidences( b.id() ) == 0,
+            "[Test] Number of block incidences should be 0" );
+    }
+}
+
+typedef void ( *test_function )( const geode::BRep& );
```

Comment:
```suggestion
using test_function = void ( * )( const geode::BRep& );
```
Is this syntax working?

Reply:
I will try